### DISCUSSION TITLE
porosity calculation very low through PNM

### DISCUSSION THREAD

**Discussion by JahanzebTariq** [PMEAL/OpenPNM#2675] [Q&A]
greeting professor @jgostick and @ma-sadeghi ... I've attached here is an issue that I am currently facing.
my image porosity is 26.7 but pnm is showing 0.3% or 0.1% porosity. what might have gone wrong here.
![image](https://user-images.githubusercontent.com/99289270/215024553-9d8027e0-cf35-4c5a-9409-8414cecd749c.png)

please address this issue. thank you

**Comment by jgostick** [comment]
You owe me a beer for debugging your code:  `(2.25**-6)` is 0.007707.  You want `(2.25e-6)` which is 0.00000225.

**Reply by JahanzebTariq** [reply]
![image](https://user-images.githubusercontent.com/99289270/215138719-c3741e69-d34c-41d2-adff-a2910613dcbf.png)

I forgot to mention that i managed to find out my mistake... but the real issue is I'm still having a very low value of porosity as compared to the one I got using the image.. is this okay...? please do reply sir

**Reply by jgostick** [reply]
PNMs will almost always have a lower porosity than the image because the pores and throats do not fill the space completely.  You can do careful adjustment to the pore and throat sizes to correct for this.  For instance, you can increase the pore diameters by some percentage. You can use the equivalent diameter instead of inscribed.  In your case, you could also use cubic throats.

**Reply by JahanzebTariq** [reply]
thank you kind sir

**Reply by JahanzebTariq** [reply]
![image](https://user-images.githubusercontent.com/99289270/215239107-8160f3a4-20eb-4acf-ab2b-1a9a3563b6ca.png)
@jgostick ... I got these values after tweaking the parameters.. but now it's high from the image.. is it okay..? or I should revert back to the previous one's

**Reply by jgostick** [reply]
You're within about 10% of the desired value, which is usually good enough for engineering calculations.

**Reply by JahanzebTariq** [reply]
![image](https://user-images.githubusercontent.com/99289270/215242309-6fa71fef-1beb-438e-b907-2ead476c7ae1.png)
@jgostick .. 1 more thing, I am encountering another issue which I never got before but now I am frequently running into this issue.. **please help me resolve this ERROR**

**Reply by jgostick** [reply]
Hmmm, that seems like an error occurring in pandas.  This usually means that your pandas needs to be upgraded, which may cause lots of other complaints.  I think the best solution is to just create a new virtual environment and install openpnm from scratch.  Like ``conda create -n name_of_new_env python=3.10`` then activate it with ``conda activate name_of_new_env``, then install openpnm with ``conda install -c conda-forge openpnm``.

**Reply by JahanzebTariq** [reply]
thank you professor... you are very helpful and kind..
